### PR TITLE
ci: add go 1.15 to tests

### PR DESCRIPTION
**Pull request by jakesyl**
Note: this requires #1619 before it can be merged.

### CODE REVIEW

**Review by onyb — APPROVED** [review]

**Comment by jakesyl** [comment]
One more thing on this. Someone with permissions (likely @Roasbeef) will have to add this as a required test once we get this merged

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: [1.13, 1.14]
+        go: [1.13, 1.14, 1.15]
```
Comment (jcvernaleo) [inline .github/workflows/go.yml:9]: I do not think we should be testing on three versions.  The policy has always been current and current-1.  So if 1.15 is current, please drop 1.13.

Comment (jakesyl) [reply .github/workflows/go.yml]: @jcvernaleo should be good to go here

**Comment by jakesyl** [comment]
looks like we'll need to remove 1.13 as a required check (and switch it to 1.15

**Comment by ipriver** [comment]
JFYI
How about to also update [README.md](https://github.com/btcsuite/btcd/blob/master/README.md#requirements) requirements version? If CI is testing only the 2 latest golang versions then there's no guarantee that the project works correctly with go version < 1.14.

**Comment by jakesyl** [comment]
@ipriver good call. I missed this

**Comment by jakesyl** [comment]
anyone know if this is still true? 
image

File: go.mod [hunk 2]
```diff
@@ -14,4 +14,4 @@ require (
 	golang.org/x/crypto v0.0.0-20200510223506-06a226fb4e37
 )
 
-go 1.12
+go 1.14
```
Comment (onyb) [inline go.mod:17]: Why not `1.15`?

**Review by onyb — CHANGES_REQUESTED** [review]
Assuming the next release will be built with Go 1.15, we need to remove `darwin-386` from `release/release.sh`. [See why](https://golang.org/doc/go1.15#darwin).

**Comment by onyb** [comment]
@jakesyl Regarding reproducible builds, the release script is still valid for Go 1.14 / 1.15 (i.e. we still need the `-trimpath` build flag):

```
➜  btcd git:(master) ✗ go build -ldflags="-s -w -buildid=" github.com/btcsuite/btcd          
➜  btcd git:(master) ✗ go tool objdump ./btcd | grep "/Users/" | wc -l                          
    2857
➜  btcd git:(master) ✗ go build -trimpath -ldflags="-s -w -buildid=" github.com/btcsuite/btcd 
➜  btcd git:(master) ✗ go tool objdump ./btcd | grep "/Users/" | wc -l                       
       0
```

Apart from the `-trimpath`, there is no ugly workaround in the release script, we should be able to remove these lines:

```
However, this wasn't _fully_ solved in `go1.13`, as the build system still includes the
directory the binary is built into the binary itself. As a result, our scripts
utilize a work around needed until `go1.13.2`.
```

File: go.mod (hunk 2 above)
Comment (ipriver) [reply go.mod:17]: I think 1.14 is fine here as it's the minimal supported version.

https://golang.org/ref/mod#go-mod-file-go
`A go directive sets the expected language version for the module.`

Comment (onyb) [reply go.mod:17]: Makes sense. 👍

**Comment by jakesyl** [comment]
@onyb all set

**Comment by jakesyl** [comment]
ah, sorry. Need to fix `release/release.sh`

**Comment by jakesyl** [comment]
_now_ I should be all set

**Comment by jakesyl** [comment]
that's new

**Review by onyb — APPROVED** [review]

**Review by jcvernaleo — APPROVED** [review]
OK

**Pull request merged**